src/System.Text.Json/porting_guide/README.md

```diff
@@ -0,0 +1,173 @@
+  - Please help contribute to this porting guide to help others onboard to the new APIs.
+  - For an overview on the objectives, constraints, and design requirements of the new APIs, please see the [roadmap](https://github.com/dotnet/corefx/blob/master/src/System.Text.Json/roadmap/README.md).
+  - If you have any feedback or feature requests, feel free to submit issues: https://github.com/dotnet/corefx/issues
+
+## Common Pitfalls, Patterns, and Tips
+
+### Using Ref Structs
+
+* The low-level `Utf8JsonReader` and `Utf8JsonWriter` are `ref structs` which means they have certain limitations (see https://aka.ms/span-safety). For example, they cannot be stored as a field on a class or struct (other than ref structs). To achieve the desired performance, these types must be `ref structs` since they need to cache the input or output `Span<byte>` (which itself is a ref struct). Furthermore, these types are mutable since they hold state. Hence, you should **pass them by ref** rather than by value. Passing them by value would result in a struct copy and the state changes would not be visible to the caller. This differs from `Newtonsoft.Json` since the `JsonTextReader`/`JsonTextWriter` you might be familiar with are classes.
+
+### Reading from a String / Stream
+
+* The `Utf8JsonReader` supports reading from a UTF-8 encoded `ReadOnlySpan<byte>` and `ReadOnlySequence<byte>` (which is the result of reading from a `PipeReader`). We currently (as of .NET Core 3.0 preview 2) do not have a convenient API to read JSON from a stream directly (either synchronously or asynchronously). For synchronous reading (especially of small payloads), you could read the JSON payload till the end of the stream into a byte array and pass that into the reader. For reading from a string (which is encoded as UTF-16), you should use the `Encoding.UTF8.GetBytes` API to first transcode the string to a UTF-8 encoded byte array, and pass that to the `Utf8JsonReader`.
+  - Code sample for async reading from a `Stream` / `PipeReader` - TBD
+
+### ReadOnlySequence, HasValueSequence, and ValueSequence
+
+* The `Utf8JsonReader` supports reading from a UTF-8 encoded `ReadOnlySpan<byte>` and `ReadOnlySequence<byte>`. If your JSON input is a **span**, each JSON element can be accessed from the `ValueSpan` property on the reader as you go through the "read loop". However, if your input is a **sequence**, some JSON elements might straddle multiple segments of the `ReadOnlySequence<byte>` and hence would not be accessible from `ValueSpan` in a contiguous memory block. Instead, whenever you have a multi-segment `ReadOnlySequence<byte>` as input, you should always poll the `HasValueSequence` property on the reader to figure out how to access the current JSON element. Here is a recommended pattern that you can follow:
```

Comment:
> Here is a recommended pattern that you can follow:
As this is a porting guide and is the current status-quo for using these types, maybe change ``can`` to ``should``?
The same sentence appears in the next part again.